src/main/java/uk/gov/justice/digital/hmpps/cmd/api/service/NotificationService.kt

```diff
@@ -146,9 +141,18 @@ class NotificationService(
     */
     private fun sendNotification(quantumId: String, notificationGroup: List<ShiftNotification>) {
         val userPreference = userPreferenceService.getOrCreateUserPreference(quantumId)
+
+        data class Key(val shiftDate: LocalDateTime, val shiftType: String)
+
+        fun ShiftNotification.toKeyDuplicates() = Key(this.shiftDate, this.shiftType)
+
+        val mostRecentNotifications = notificationGroup
+                .groupBy { it.toKeyDuplicates() }
+                .map { (key, value) -> value.sortedByDescending { it.shiftModified }.elementAt(0) }
```

Comment:
Would this be cleaner with a minBy/maxBy?

Reply:
Stop it, that's an inbuilt method 🤩